lib/kitchen/provisioner/chef_zero.rb

```diff
@@ -56,47 +55,40 @@ def prepare_command
         # the installed omnibus package. Yep, this is funky :)
         cmd = <<-PREPARE.gsub(/^ {10}/, "")
           #{chef_client_zero_env(:export)}
-          if ! #{sudo(ruby_bin.join("gem"))} list chef-zero -i >/dev/null; then
+          if ! #{shell.sudo(ruby_bin.join("gem"))} list chef-zero -i >/dev/null; then
             echo ">>>>>> Attempting to use chef-zero with old version of Chef"
             echo "-----> Installing chef zero dependencies"
-            #{sudo(ruby_bin.join("gem"))} install chef --no-ri --no-rdoc --conservative
+            #{shell.sudo(ruby_bin.join("gem"))} install chef --no-ri --no-rdoc --conservative
           fi
         PREPARE
 
-        Util.wrap_command(cmd, shell)
+        shell.wrap_command(cmd)
       end
 
       # (see Base#run_command)
       def run_command
         level = config[:log_level] == :info ? :auto : config[:log_level]
-        chef_client_bin = sudo(config[:chef_client_path])
-
-        cmd = modern? ? "#{chef_client_bin} --local-mode" : shim_command
-        args = [
-          "--config #{config[:root_path]}/client.rb",
-          "--log_level #{level}",
-          "--force-formatter",
-          "--no-color"
-        ]
-        if config[:chef_zero_host]
-          args <<  "--chef-zero-host #{config[:chef_zero_host]}"
-        end
-        if config[:chef_zero_port]
-          args <<  "--chef-zero-port #{config[:chef_zero_port]}"
-        end
-        if config[:json_attributes]
-          args << "--json-attributes #{config[:root_path]}/dna.json"
-        end
-        if config[:log_file]
-          args << "--logfile #{config[:log_file]}"
-        end
-
-        # TODO: We definitely need to put more logic on this.
-        Util.wrap_command([cmd, *args].join(" "), shell)
+        cmd = [modern? ? "#{shell.sudo(config[:chef_client_path])} --local-mode" : shim_command]
+        cmd << "--config #{config[:root_path]}/client.rb"
+        cmd << "--log_level #{level}"
+        cmd << "--force-formatter"
+        cmd << "--no-color"
+        add_argument(cmd, :chef_zero_host, "--chef-zero-host %config%")
+        add_argument(cmd, :chef_zero_port, "--chef-zero-port %config%")
+        add_argument(cmd, :json_attributes, "--json-attributes #{config[:root_path]}/dna.json")
+        add_argument(cmd, :log_file, "--logfile %config%")
+
+        shell.wrap_command(cmd.join(" "))
       end
 
       private
 
+      def add_argument(line, config_key, arg)
```

Comment:
I'm gonna try to do this a better way. It gets the job done and appeases rubocop but is just kinda weird.